src/client/entity-editor/common/entity.js

```diff
@@ -48,6 +48,7 @@ function Entity(
 				disambiguation &&
 				<span className="disambig"><small>({disambiguation})</small></span>
 			}
+			<span className="text-muted small">{language}</span>
```

Comment:
I am not convinced you should add this here. I understand the reason, and it's a good reason, but the Entity component is not used only in this select but also in other components.

I think for now seeing the extra details only in the select options is sufficient.

Reply:
What I mean is the language should not be added in this component.

Reply:
No, I would just not add the language.